core/src/main/java/org/apache/accumulo/core/util/threads/Threads.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.util.threads;
+
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.util.OptionalInt;
+
+public class Threads {
+
+  public static Runnable createNamedRunnable(String name, Runnable r) {
+    return new NamedRunnable(name, r);
+  }
+
+  public static Runnable createNamedRunnable(String name, OptionalInt priority, Runnable r) {
+    return new NamedRunnable(name, priority, r);
+  }
+
+  private static final UncaughtExceptionHandler UEH = new AccumuloUncaughtExceptionHandler();
+
+  public static Thread createThread(String name, Runnable r) {
+    return createThread(name, OptionalInt.empty(), r);
+  }
+
+  public static Thread createThread(String name, OptionalInt priority, Runnable r) {
+    Thread thread = null;
+    if (r instanceof NamedRunnable) {
+      NamedRunnable nr = (NamedRunnable) r;
+      thread = new Thread(r, name);
+      if (nr.getPriority().isPresent()) {
+        thread.setPriority(nr.getPriority().getAsInt());
+      } else if (priority.isPresent()) {
+        thread.setPriority(priority.getAsInt());
+      }
+    } else {
+      thread = new Thread(r, name);
+      if (priority.isPresent()) {
+        thread.setPriority(priority.getAsInt());
+      }
+    }
```

Comment:
Just a nit, but it doesn't appear that the initialization of `thread` is any different based on the type of Runnable passed in. Why not move the initialization up? Maybe something like:
```suggestion
    Thread thread = new Thread(r, name);
    if (r instanceof NamedRunnable) {
      NamedRunnable nr = (NamedRunnable) r;
      if (nr.getPriority().isPresent()) {
        priority = nr.getPriority();
      }
    }
    if (priority.isPresent()) {
      thread.setPriority(priority.getAsInt());
    }
```

Reply:
I made this change (in the next commit), but I modified your suggestion such that if the priority is set in the NamedRunnable it is not overridden by the priority parameter.

Reply:
> I made this change (in the next commit), but I modified your suggestion such that if the priority is set in the NamedRunnable it is not overridden by the priority parameter.

I think my suggestion would have done that too, since it overwrites the priority parameter on line 45, but if you missed that then my suggestion was too subtle so it's best not to use it. :)

Reply:
Ah, I see it now. My change should be functionally equivalent.